### PR TITLE
fuzzgen: Generate Tail Calls

### PR DESCRIPTION
**Pull request by afonso360**
👋 Hey,

This PR is a follow up to  #6635 and it allows fuzzgen to generate `return_call` and `return_indirect_call` instructions. These are modeled as any other control flow instruction however they have a bunch of restrictions before we are actually able to insert them.

As with the other control flow instructions, we try to include as much info as possible in the CFG before inserting it. It doesn't seem as relevant with tail calls as it is with blocks, but I figured it is still worth it. (The alternative would be to pick the tail call target at instruction insertion time, but I think that would actually make this slightly more complicated)

I'm opening this as a draft since it has found some issues and I can't run it for too long before it crashes, and I'm not entirely sure it's generating correct code under all inputs. Just leaving this open if anyone wants to test with it (@fitzgen  👀 ).

### CODE REVIEW

**Comment by fitzgen** [comment]
Nice! The last commit LGTM, thanks for putting this up. Going to look into the issues you found now.

**Comment by afonso360** [comment]
Fuzzed for around 2 hours with fuzzgen and it all seems stable! 🥳

File: cranelift/fuzzgen/src/function_generator.rs [hunk 1]
```diff
@@ -68,27 +69,32 @@ fn insert_opcode(
...
         sig.params.iter().map(|abi_param| abi_param.value_type),
     )?;
 
-    let call = if opcode == Opcode::Call {
-        builder.ins().call(func_ref, &actuals)
-    } else {
-        let addr_ty = args[0];
-        let addr = builder.ins().func_addr(addr_ty, func_ref);
-        builder.ins().call_indirect(sig_ref, addr, &actuals)
+    let addr_ty = I64;
```
Comment (fitzgen) [inline cranelift/fuzzgen/src/function_generator.rs:85]: Can we maybe factor this out to a helper that checks `cfg!(target_pointer_width = "64")` and then has an `else` that panics with a helpful message? Just to future proof this a little bit.

**Review by fitzgen — APPROVED** [review]
Looks great! Thanks a ton!

File: cranelift/fuzzgen/src/function_generator.rs (hunk 1 above)
Comment (afonso360) [reply cranelift/fuzzgen/src/function_generator.rs]: Turns out we can already query the ISA's pointer type directly, so I've done just that.

**Pull request merged**